File: cpp/codeforces/interesting_drink/interesting_drink.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int num_shops, num_days;
    cin >> num_shops;
    vector<int> shops;
    for(int i = 0; i < num_shops; i++) {
        int x;
        cin >> x;
        shops.push_back(x);
    }
    sort(begin(shops), end(shops));
    cin >> num_days;
    for(int i = 0; i < num_days; i++) {
        int today_money;
        cin >> today_money;
        int lo = -1;
        int hi = shops.size();
        while(lo + 1 < hi) {
            int mid = (int) ((lo + hi) / 2);
            if(shops[mid] <= today_money) lo = mid;
            else hi = mid;
        }
        cout << lo + 1 << endl;
    }

}

File: python/codeforces/interesting_drink/interesting_drink.py
num_shops = int(input())
shop_costs = sorted(map(int, input().split()))
num_days = int(input())

sol = list()

for d in range(num_days):
    today_money = int(input())
    lo = -1
    hi = len(shop_costs)
    while lo + 1 < hi:
        mid = int((lo + hi) / 2)
        if shop_costs[mid] <= today_money: lo = mid
        else: hi = mid
    sol += f"{lo + 1}"
    sol += "\n"
print("".join(sol[:-1]))
